scripts/compare-logits.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import requests
+import json
+from pathlib import Path
+
+
+DESCRIPTION = """
+Compare logits between llama.cpp and another inference engine.
+
+Example usage:
+    Step 1: Dump logits from two different endpoints
+        python scripts/compare-logits.py dump logits_llama.log http://localhost:8080/completions
+        python scripts/compare-logits.py dump logits_other.log http://other-engine:8000/completions
+
+        (optionally, you can add --api-key <key> if the endpoint requires authentication)
+
+    Step 2: Compare the dumped logits
+        python scripts/compare-logits.py compare logits_llama.log logits_other.log report.md
+"""
+
+
+INPUT_PROMPT = """
+You are an advanced AI assistant capable of using tools to gather information, perform calculations, or execute tasks. Always think step by step before responding. If a user's query requires external data, computation, or actions beyond your internal knowledge, use the appropriate tools via function calls.
+
+### Tool Call Format:
+When you need to use a tool, output the call in this exact XML-inspired format. Include the opening and closing tags. Do not escape arguments; they will be parsed as plain text.
+
+You can make multiple calls in one go by placing them one after another.
+""".strip()
+
+
+
+def dump_logits(endpoint: str, output_path: Path, api_key = None):
+    print(f"Dumping logits to {output_path} from endpoint {endpoint}...")
+    words = INPUT_PROMPT.split(' ')
+    curr_text = ""
+    with output_path.open("w") as f:
+        for i, w in enumerate(words):
+            curr_text += (w + " ")
+            payload = {
+                "prompt": curr_text.strip(),
+                "temperature": 0.0,
+                "top_k": 1,
+                "max_tokens": 1,
+                "logprobs": 1,
+                "stream": False,
```

Comment:
Just note that I'm no longer use the `echo` option because the support is hit-or-missed across frameworks